src/Integration.Vsix/SonarLintNotificationsPackage.cs

```diff
@@ -65,26 +65,23 @@ protected override void Initialize()
             notifications = new SonarQubeNotificationService(sonarqubeService,
                 new NotificationIndicatorViewModel(), new TimerWrapper { Interval = 60000 }, sonarLintOutput);
 
+            notificationIcon.DataContext = notifications.Model;
+
             activeSolutionBoundTracker = this.GetMefService<IActiveSolutionBoundTracker>();
             activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
         }
 
         private void OnSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs binding)
         {
-            if (notificationIcon == null)
-            {
-                notificationIcon = new NotificationIndicator();
-                notificationIcon.DataContext = notifications.Model;
-                VisualStudioStatusBarHelper.AddStatusBarIcon(notificationIcon);
-            }
-
             if (binding.IsBound)
             {
+                VisualStudioStatusBarHelper.AddStatusBarIcon(notificationIcon);
```

Comment:
What was the root cause of the notifications not appearing?
It looks like the old version of the code  added the notification icon once, and the used data-binding to the IsIconVisible property of the model on/off. Presumably some of that code is now redundant and can be removed?

Reply:
Good question.

The problem is that the icon was attached to a visual tree, which gets removed when solution is closed - new solution gets new visual tree, which means the icon needs to be re-added.

We still us IsIconVisible. At the point when the icon is added, we don't know yet if it is going to be visible (e.g. if server supports notifications). This could be refactored so we only add the icon when we know it is, I don't think it's worth it at this stage.

Reply:
Makes sense, thanks for the info.